### PR TITLE
Giraffe autorelocate

### PR DESCRIPTION
**Pull request by jranks123**
- Implements auto relocation for visitors to the contributions page, based on their IP.
- Sends European, US, AU and UK visitors to their specific currency pages, defaults to US dollars

### CODE REVIEW

File: frontend/app/controllers/Giraffe.scala [hunk 1]
```diff
@@ -110,3 +118,15 @@ object Giraffe extends Controller {
     })
   }
 }
+
+object MakeURL {
```
Comment (paulbrown1982) [inline frontend/app/controllers/Giraffe.scala]: Have you seen the netaporter Scala Uri API? 

import com.netaporter.uri.dsl._
import com.netaporter.uri.Uri

With that you can do something like this:

var x = Uri.parse(request.uri).withScheme("https")
val url = x.copy(pathParts = Seq(PathPart(countryGroup.id)) ++ x.pathParts)

Comment (jranks123) [inline frontend/app/controllers/Giraffe.scala]: Thanks Paul, this is much better

**Comment by paulbrown1982** [comment]
👍

**Comment by markjamesbutler** [comment]
:shipit:

**Pull request merged**